sdk/identity/azure-identity/src/main/java/com/azure/identity/EnvironmentCredential.java

```diff
@@ -56,19 +59,29 @@
         String certPath = configuration.get(Configuration.PROPERTY_AZURE_CLIENT_CERTIFICATE_PATH);
         String username = configuration.get(Configuration.PROPERTY_AZURE_USERNAME);
         String password = configuration.get(Configuration.PROPERTY_AZURE_PASSWORD);
+        logAvailableEnvironmentVariables(clientId, tenantId, clientSecret, certPath, username, password);
         if (verifyNotNull(clientId)) {
             if (verifyNotNull(tenantId, clientSecret)) {
+                logger.info("Azure Identity => EnvironmentCredential invoking ClientSecretCredential");
                 targetCredential = new ClientSecretCredential(tenantId, clientId, clientSecret, identityClientOptions);
             } else if (verifyNotNull(tenantId, certPath)) {
+                logger.info("Azure Identity => EnvironmentCredential invoking ClientCertificateCredential");
                 targetCredential = new ClientCertificateCredential(tenantId, clientId, certPath,
                         null, identityClientOptions);
             } else if (verifyNotNull(username, password)) {
+                logger.info("Azure Identity => EnvironmentCredential invoking UsernamePasswordCredential");
                 targetCredential = new UsernamePasswordCredential(clientId,
                         tenantId,
                         username,
                         password,
                         identityClientOptions);
+            } else {
+                logger.error("Azure Identity => ERROR in EnvironmentCredential: Not enough environment variables "
```

Comment:
Should this be "Required Environment variables not found to determine an authentication credential?

Reply:
Some required environment variables may have been found. so "required environment variables not found" is not accurate. I've added more detail to this error message if that makes it more readable to you.